x-pack/solutions/search/plugins/search_getting_started/public/components/header/add_data_button.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+import { EuiButton, EuiContextMenuPanel, EuiContextMenuItem, EuiPopover } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '../../hooks/use_kibana';
+
+export const AddDataButton: React.FC = () => {
+  const {
+    services: { application },
+  } = useKibana();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const onButtonClick = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const items = [
+    <EuiContextMenuItem
+      key="upload"
+      icon="exportAction"
+      onClick={() => {
+        closePopover();
+        application.navigateToUrl('/app/ml/filedatavisualizer');
+      }}
+    >
+      {i18n.translate('xpack.search.gettingStarted.addDataButton.uploadFile', {
+        defaultMessage: 'Upload a file',
+      })}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="sample"
+      icon="folderOpen"
+      onClick={() => {
+        closePopover();
+        application.navigateToUrl('/app/home#/tutorial_directory/sampleData');
+      }}
+    >
+      {i18n.translate('xpack.search.gettingStarted.addDataButton.browseSampleData', {
+        defaultMessage: 'Browse sample data sets',
+      })}
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="empty"
+      icon="indexOpen"
+      onClick={() => {
+        closePopover();
+        application.navigateToUrl('/app/elasticsearch/indices/create');
```

Comment:
If I understand correctly we want this changes to work for self hosted as well? if yes, we may want below changes 
```suggestion
        application.navigateToUrl(`http.basePath.prepend('/app/elasticsearch/indices/create'))`;
```